core/src/main/scala/org/apache/spark/storage/BlockInfoManager.scala

```diff
@@ -150,6 +150,12 @@ private[storage] class BlockInfoManager extends Logging {
    */
   private[this] val blockInfoWrappers = new ConcurrentHashMap[BlockId, BlockInfoWrapper]
 
+  /**
+   * Record visible rdd blocks stored in the block manager, entries will be removed
+   * by [[removeBlock()]]
+   */
+  private[spark] val visibleRDDBlocks = ConcurrentHashMap.newKeySet[RDDBlockId]
```

Comment:
It's weird to me that we use the opposite visibility tracking logic between the driver and executor. Couldn't we use the same "invisible" logic at the executor too?